### PR TITLE
chore: github actions workflow for tests / visual regressions

### PR DESCRIPTION
**Pull request by holloway**
* Add CI workflow for tests (vitest, TS, lint)
* Moves generated JSON into the `generated` directory
* Renames `vis-reg-test.yml` to the more readable `visual-regression-test.yml`
* Temporarily commits the api client to Git (see comment below)

### CODE REVIEW

**Comment by holloway** [comment]
@jennifer-richards now that we've got unit tests I've added a workflow to run those, TS, and linting on CI.

Because the Nuxt build and TS require the generated API client we'll need to have the client on CI. It would be best to generate that on CI so that we have current types, but I've temporarily commited the API client (which is a single file) to test this workflow.

**Review by jennifer-richards — DISMISSED** [review]

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          working-directory: './client'
+  
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+            node-version: 23.x
```
Comment (NGPixel) [inline .github/workflows/tests.yml:19]: I would avoid using non-LTS versions in anything unless absolutely necessary.

**Review by NGPixel — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (holloway) [reply .github/workflows/tests.yml]: Ah it was the quickest fix for this... I'll look for a workaround
https://github.com/ietf-tools/rfced-www/actions/runs/12311686063/job/34362303820

Comment (holloway) [reply .github/workflows/tests.yml]: Done

**Review by jennifer-richards — APPROVED** [review]
lgtm

**Review by NGPixel — APPROVED** [review]

**Pull request merged**